### PR TITLE
Add ability to configure the NotBefore property of certificates in role api

### PR DESCRIPTION
**Pull request by sk4ry**
Correction of #4639

### CODE REVIEW

**Comment by sk4ry** [comment]
Close pull request, need to relaunch CI

File: builtin/logical/pki/path_roles.go [hunk 1]
```diff
@@ -286,6 +286,11 @@ for "generate_lease".`,
 				Type:        framework.TypeBool,
 				Description: `Mark Basic Constraints valid when issuing non-CA certificates.`,
 			},
+			"not_before": &framework.FieldSchema{
+				Type:        framework.TypeInt,
```
Comment (chrishoffman) [inline builtin/logical/pki/path_roles.go]: Can you switch this to TypeDuration which will require some other changes in storage.  See max_ttl for an example of a TypeDuration field.

File: builtin/logical/pki/path_roles.go [hunk 2]
```diff
@@ -286,6 +286,11 @@ for "generate_lease".`,
...
+				Default:     30,
```
Comment (chrishoffman) [inline builtin/logical/pki/path_roles.go]: Would you mind removing the default from both the backend and frontend definitions?  If this value is the default zero value, it should just ignore this field.

File: ui/app/models/role-pki.js [hunk 3]
```diff
@@ -104,6 +104,11 @@ export default DS.Model.extend({
   basicConstraintsValidForNonCA: attr('boolean', {
     label: 'Mark Basic Constraints valid when issuing non-CA certificates.',
   }),
+  notBeforeDuration: attr({
+    label: 'Not Before Duration',
```
Comment (meirish) [inline ui/app/models/role-pki.js]: You don't have to have this line as it will "humanize" the attribute name, but fine to leave it in.

**Pull request merged**